pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -215,7 +241,7 @@ func (m *MigCluster) GetClient(c k8sclient.Client) (compat.Client, error) {
 		return nil, err
 	}
 	if rwStarted {
-		clientMap[m.UID] = compatClient
+		clientMap.Set(m.UID, compatClient)
```

Comment:
_I know this PR isn't aiming to solve the problem I'm about to describe, but I think we should fix this before merging cached client stuff into master:_

We should also be keeping an in-memory map of clients for each cluster when the user _isn't_ doesn't have cached client turned on, because a huge part of the cost is actually just client creation. I've observed that compat client creation is a lot more expensive than compat client usage. So even non-cached client users could benefit from this change, and our performance will be better before the cached client is populated.

Will add this as a work item to the original PR.